modules/affile/affile-dest.c

```diff
@@ -162,8 +169,10 @@ affile_dw_reopen(AFFileDestWriter *self)
 
       LogTransport *transport = file_opener_construct_transport(self->owner->file_opener, fd);
 
+      FileReopener reopener = { .request_reopen = _file_reopen_cb, .cookie = self};
       proto = file_opener_construct_dst_proto(self->owner->file_opener, transport,
-                                              &self->owner->writer_options.proto_options.super);
+                                              &self->owner->writer_options.proto_options.super, self->owner->super.super.super.signal_slot_connector, self->filename,
```

Comment:
the slot is probably better placed in the FileOpener constructor so that if an implementation of construct_dst_proto needs it it can find it in self and we wouldn't need to change the prototype here.

I guess this only applies to regular files and not so much to /proc/kmsg or named pipes.